p2p/src/conn.rs

```diff
@@ -177,14 +184,24 @@ where
 	let (close_tx, close_rx) = mpsc::channel();
 	let (error_tx, error_rx) = mpsc::channel();
 
+	// Counter of number of bytes received
+	let received_bytes = Arc::new(Mutex::new(0));
+
 	stream
 		.set_nonblocking(true)
 		.expect("Non-blocking IO not available.");
-	poll(stream, handler, send_rx, error_tx, close_rx);
+	poll(
+		stream,
+		handler,
+		send_rx,
+		error_tx,
+		close_rx,
+		received_bytes.clone(),
+	);
 
 	Tracker {
 		sent_bytes: Arc::new(Mutex::new(0)),
-		received_bytes: Arc::new(Mutex::new(0)),
+		received_bytes: received_bytes.clone(),
```

Comment:
What do you think about using `RWLock` for both counters?

Reply:
It seems more logical to use single-producer multiple-consumer enabled lock, yes.